# **Image** **Classification**

In [ ]:
!pip install ultralytics

# **Code for Image Classification using YOLOV8**

In [13]:
from ultralytics import YOLO
from PIL import Image

# Load a pretrained YOLOv8 classification model
model = YOLO('yolov8n-cls.pt')  # nano model

def classify_image(img_path):
    # Load the image
    img = Image.open(img_path)

    # Perform classification
    results = model(img)[0]

    # Print top 5 predictions
    for i, (cls, conf) in enumerate(zip(results.names.values(), results.probs.top5)):
        print(f"{i+1}: {cls} ({conf:.2f})")


In [14]:
# Example usage
image_path = '/content/plane.jpeg'
classify_image(image_path)


0: 224x224 airliner 1.00, warplane 0.00, wing 0.00, space_shuttle 0.00, airship 0.00, 7.0ms
Speed: 4.6ms preprocess, 7.0ms inference, 0.1ms postprocess per image at shape (1, 3, 224, 224)
1: tench (404.00)
2: goldfish (895.00)
3: great_white_shark (908.00)
4: tiger_shark (812.00)
5: hammerhead (405.00)


In [17]:
# Example usage
image_path2 = '/content/german.jpeg'
classify_image(image_path2)


0: 224x224 German_shepherd 0.97, malinois 0.02, Norwegian_elkhound 0.00, dhole 0.00, kelpie 0.00, 6.0ms
Speed: 3.5ms preprocess, 6.0ms inference, 0.1ms postprocess per image at shape (1, 3, 224, 224)
1: tench (235.00)
2: goldfish (225.00)
3: great_white_shark (174.00)
4: tiger_shark (274.00)
5: hammerhead (227.00)


# ***100% prediction for Aeroplane and 97% pred. for German_Sphered***